airbyte-integrations/connectors/source-dixa/source_dixa/source.py

```diff
@@ -0,0 +1,128 @@
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+from abc import ABC
+from datetime import date, datetime, timedelta
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+
+
+DATE_FORMAT = "%Y-%m-%d"
+
+
+class DixaStream(HttpStream, ABC):
+    url_base = "https://exports.dixa.io/v1/"
+    date_format = DATE_FORMAT
+    max_days_in_query = 31  # see https://support.dixa.help/en/articles/174-export-conversations-via-api
```

Comment:
Would it be wise to make this a little smaller e.g: a week or less? Since the endpoint doesn't have any pagination strategy, we could end up making gigantic requests otherwise

Reply:
Good point. The reason I pushed it to the highest possible value was that the rate limit is quite low (10 requests per 60 seconds). With batches of 7 days, the connector would spend around 5 minutes sleeping when importing a whole year. With batches of one day, that number jumps to >30 minutes... Not ideal.

This wouldn't be that big of a problem if incremental is an option, but still worth considering. What do you think of making the batch size a parameter of the connector?

Reply:
making it a param is a great idea. It can be an optional parameter with a default value of 31.